File: android/app/src/main/java/fr/jarvis/companion/JarvisBootReceiver.java
package fr.jarvis.companion;

import android.Manifest;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;

/** Restaure la présence GPS ; le micro exige une action visible après un redémarrage. */
public final class JarvisBootReceiver extends BroadcastReceiver {
    @Override public void onReceive(Context context, Intent intent) {
        JarvisNotifications.createChannels(context);
        boolean locationEnabled = JarvisSettings.preferences(context)
                .getBoolean(JarvisSettings.PREF_LOCATION, false);
        if (locationEnabled
                && context.checkSelfPermission(Manifest.permission.ACCESS_FINE_LOCATION)
                == PackageManager.PERMISSION_GRANTED
                && context.checkSelfPermission(Manifest.permission.ACCESS_BACKGROUND_LOCATION)
                == PackageManager.PERMISSION_GRANTED) {
            context.startForegroundService(new Intent(context, JarvisLocationService.class));
        }

        boolean wakeEnabled = JarvisSettings.preferences(context)
                .getBoolean(JarvisSettings.PREF_WAKE, false);
        if (wakeEnabled) {
            JarvisNotifications.show(
                    context,
                    JarvisNotifications.DEFAULT,
                    "Réactiver l'écoute JARVIS",
                    "Android exige une ouverture de l'application après le redémarrage",
                    false
            );
        }
    }
}

File: android/app/src/main/kotlin/fr/jarvis/companion/receivers/JarvisBootReceiver.kt
package fr.jarvis.companion.receivers

import android.Manifest
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import fr.jarvis.companion.core.sync.LocationSyncWorker
import fr.jarvis.companion.data.JarvisSettings
import fr.jarvis.companion.notifications.JarvisNotifications
import fr.jarvis.companion.services.JarvisLocationService

/** Restaure GPS et sync location après redémarrage. */
class JarvisBootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        if (!isSupportedBootAction(intent?.action)) return
        JarvisNotifications.createChannels(context)

        val plan = bootRecoveryPlan(
            hasNativeToken = JarvisSettings.nativeToken(context).isNotEmpty(),
            locationEnabled = JarvisSettings.isLocationEnabled(context),
            hasLocationPermission = context.checkSelfPermission(
                Manifest.permission.ACCESS_FINE_LOCATION,
            ) == PackageManager.PERMISSION_GRANTED,
            wakeWordEnabled = JarvisSettings.isWakeWordEnabled(context),
        )

        if (plan.scheduleLocationSync) {
            LocationSyncWorker.schedule(context)
            LocationSyncWorker.enqueueNow(context)
        }

        if (plan.startLocationService) {
            context.startForegroundService(Intent(context, JarvisLocationService::class.java))
        }

        if (plan.showWakeWordReminder) {
            JarvisNotifications.show(
                context,
                JarvisNotifications.DEFAULT,
                "Réactiver l'écoute JARVIS",
                "Android exige une ouverture de l'application après le redémarrage",
            )
        }
    }
}
